codersdk/workspaceagents.go

```diff
@@ -421,6 +421,15 @@ type WorkspaceAgentDevcontainer struct {
 	Error string `json:"error,omitempty"`
 }
 
+func (d WorkspaceAgentDevcontainer) Equals(other WorkspaceAgentDevcontainer) bool {
+	return d.ID == other.ID &&
+		d.Name == other.Name &&
+		d.WorkspaceFolder == other.WorkspaceFolder &&
+		d.Status == other.Status &&
+		d.Dirty == other.Dirty &&
+		d.Error == other.Error
```

Comment:
We should also check container and agent, so something like this?

```suggestion
		d.Error == other.Error &&
		(d.Container == nil && other.Container == nil || (d.Container != nil && other.Continer != nil && d.Container.ID == other.Container.ID)) &&
		(d.Agent == nil && ... d.Agent.ID == ...
```

Reply:
Oh silly me I only read half the sentence 🤦‍♀️